src/main/java/org/testng/internal/ClassHelper.java

```diff
@@ -182,7 +182,7 @@ private static void logClassNotFoundError(String className, Exception ex) {
     }
 
     Class<?> parent = clazz.getSuperclass();
-    while (null != parent) {
+    while (null != parent && (!Object.class.equals(parent))) {
```

Comment:
@juherr .. If we use a `do..while` then we will trigger a NPE because `getAvailableMethods()` ends up calling `extractMethods()` which invokes methods on the `parent` class reference. So it has to be a entry controlled loop and not an exit controlled loop.